library/psa_crypto.c

```diff
@@ -3091,7 +3091,7 @@ static psa_status_t psa_cipher_setup( psa_cipher_operation_t *operation,
 
     /* The requested algorithm must be one that can be processed by cipher. */
     if( ! PSA_ALG_IS_CIPHER( alg ) )
-        return( PSA_ERROR_INVALID_ARGUMENT );
+        return( PSA_ERROR_NOT_SUPPORTED );
```

Comment:
For mac algorithm, like psa_mac_compute(), the spec also defines return PSA_ERROR_NOT_SUPPORTED if alg is not supported or is not a MAC algorithm. Not only for cipher.

Is there any page to record this spec bug? since for some users, we just implement our code follows the spec and don't investigate whether the definition is proper or not.

Reply:
It's tracked at https://github.com/ARMmbed/psa-crypto-api/issues/492 (private link) and we (mostly Andrew) are working on it.